provider/pkg/clients/clients.go

```diff
@@ -164,23 +166,25 @@ func (dcs *DynamicClientSet) searchKindInGVResources(gvResources *v1.APIResource
 // IsNamespacedKind checks if a given GVK is namespace-scoped. Known GVKs are compared against a static lookup table.
 // For GVKs not in the table, attempt to look up the GVK from the API server. If the GVK cannot be found, a
 // NoNamespaceInfoErr is returned.
-func IsNamespacedKind(gvk schema.GroupVersionKind, clientSet *DynamicClientSet) (bool, error) {
+func IsNamespacedKind(gvk schema.GroupVersionKind, disco discovery.DiscoveryInterface) (bool, error) {
 	gv := gvk.GroupVersion().String()
 	if strings.Contains(gv, "core/v1") {
 		gv = "v1"
 	}
 
-	kind := strings.TrimSuffix(gvk.Kind, "Patch") // Check using the underlying kind for Patch resources
-	if known, namespaced := kinds.Kind(kind).Namespaced(); known {
-		return namespaced, nil
+	if kinds.KnownGroupVersions.Has(gvk.GroupVersion().String()) {
+		kind := strings.TrimSuffix(gvk.Kind, "Patch") // Check using the underlying kind for Patch resources
```

Comment:
I believe it unnecessary to trim the `Patch` suffix, but is also harmless because its applied only to the well-known kinds. cc @lblackstone